challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |str| str.capitalize()}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.find_all{|str| str == 'dog'}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.find_all{|str| str != 'dog'}
```

Comment:
Try to use a new `Enumerable` method for each solution.  So L7 and 23 could be switched out for new methods.  Want to give that a try?